docs/api/python/gluon.md

```diff
@@ -21,58 +21,379 @@ in Python and then deploy with symbolic graph in C++ and Scala.
+
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    Conv1D
+    Conv2D
+    Conv3D
+    Conv1DTranspose
+    Conv2DTranspose
+    Conv3DTranspose
+```
+
+
+
+### Pooling Layers
 
-## Containers
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    MaxPool1D
+    MaxPool2D
+    MaxPool3D
+    AvgPool1D
+    AvgPool2D
+    AvgPool3D
+    GlobalMaxPool1D
+    GlobalMaxPool2D
+    GlobalMaxPool3D
+    GlobalAvgPool1D
+    GlobalAvgPool2D
+    GlobalAvgPool3D
+```
+
+
+
+## Recurrent Layers
+
+```eval_rst
+.. currentmodule:: mxnet.gluon.rnn
+```
+
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    RecurrentCell
+    RNN
+    LSTM
+    GRU
+    RNNCell
+    LSTMCell
+    GRUCell
+    SequentialRNNCell
+    BidirectionalCell
+    DropoutCell
+    ZoneoutCell
+    ResidualCell
+```
+
+
+## Trainer
 
 ```eval_rst
 .. currentmodule:: mxnet.gluon
-.. autoclass:: mxnet.gluon.Block
-    :members:
 
-    .. automethod:: forward
-.. autoclass:: mxnet.gluon.HybridBlock
-    :members:
+.. autosummary::
+    :nosignatures:
 
-    .. automethod:: hybrid_forward
+    Trainer
 ```
 
-## Neural Network Layers
+
+## Loss functions
 
 ```eval_rst
-.. currentmodule:: mxnet.gluon.nn
+.. currentmodule:: mxnet.gluon.loss
 ```
 
-### Containers
 
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    L2Loss
+    L1Loss
+    SoftmaxCrossEntropyLoss
+    KLDivLoss
+```
+
+## Utilities
 
 ```eval_rst
-.. currentmodule:: mxnet.gluon.nn
+.. currentmodule:: mxnet.gluon.utils
+```
 
-    .. automethod:: __call__
-.. autoclass:: mxnet.gluon.nn.Sequential
-    :members:
-.. autoclass:: mxnet.gluon.nn.HybridSequential
-    :members:
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    split_data
+    split_and_load
+    clip_global_norm
 ```
 
+## Data
 
-### Basic Layers
+```eval_rst
+.. currentmodule:: mxnet.gluon.data
+```
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    Dataset
+    ArrayDataset
+    RecordFileDataset
+    ImageRecordDataset
+```
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    Sampler
+    SequentialSampler
+    RandomSampler
+    BatchSampler
+```
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    DataLoader
+```
+
+### Vision
+
+```eval_rst
+.. currentmodule:: mxnet.gluon.data.vision
+```
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    MNIST
+    CIFAR10
+```
+
+## Model Zoo
+
+Model zoo provides pre-defined and pre-trained models to help bootstrap machine learning applications.
+
+### Vision
+
+```eval_rst
+.. currentmodule:: mxnet.gluon.model_zoo.vision
+```
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    get_model
+```
+
+#### ResNet
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    resnet18_v1
+    resnet34_v1
+    resnet50_v1
+    resnet101_v1
+    resnet152_v1
+    resnet18_v2
+    resnet34_v2
+    resnet50_v2
+    resnet101_v2
+    resnet152_v2
+```
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    ResNetV1
+    ResNetV2
+    BasicBlockV1
```

Comment:
Users can use the blocks to construct their own ResNet outside of predefined models (resnet1001 e.g.)